### PR TITLE
otelcol-contrib 0.92.0 (new formula)

### PR DESCRIPTION
**Pull request by edigaryev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by p-linnane** [comment]
Thanks for the PR @edigaryev. I'm a little concerned about this notice on the upstream repo:
>⚠️ Important note: Git tags in this repository may change at any time to fix any issues found during a release. They are only meant to trigger Github releases and should not be relied upon.

If this is upstream's policy, this formula will regularly break if they alter tags after release. The [manual for git](https://git-scm.com/docs/git-tag#_on_re_tagging) states that re-tagging is "the insane thing" to do, and we agree with that. It becomes very hard to rule out malicious behavior with checksums changing, so I'm not sure we can accept this unfortunately.

**Comment by SMillerDev** [comment]
It also violates: https://docs.brew.sh/Acceptable-Formulae#stable-versions

**Comment by mx-psi** [comment]
> Thanks for the PR @edigaryev. I'm a little concerned about this notice on the upstream repo:
> 
> > ⚠️ Important note: Git tags in this repository may change at any time to fix any issues found during a release. They are only meant to trigger Github releases and should not be relied upon.
> 
> If this is upstream's policy, this formula will regularly break if they alter tags after release. The [manual for git](https://git-scm.com/docs/git-tag#_on_re_tagging) states that re-tagging is "the insane thing" to do, and we agree with that. It becomes very hard to rule out malicious behavior with checksums changing, so I'm not sure we can accept this unfortunately.

Let me clarify a few points, speaking as the person that wrote that warning:
- The opentelemetry-collector-releases repository does not have any of the code from the OpenTelemetry Collector, only the release pipeline code
- We never re-tag the https://github.com/open-telemetry/opentelemetry-collector-contrib or https://github.com/open-telemetry/opentelemetry-collector repositories, where the Go code lives. If there is an issue, we just release a new bugfix version.
- Once artifacts have been released over at opentelemetry-collector-releases, we will never remove them, opting to make a new Github Release instead

In summary: the warning is meant to express that if an issue happens from the tagging up until the Github release happens, we can change the release pipeline code to make sure the release happens smoothly; it does **not** mean that we will republish any of the binaries published via Github releases

I am happy to clarify this on the README from our repository if necessary. I am also happy to revisit the policy if necessary, but to be clear the Git tags on that repository are in practical terms meaningless.
